### PR TITLE
Upgrade to actix web 2

### LINKED ISSUE #394: Upgrade to actix web 2.x
It's probably worth making time to upgrade to 2.x before we get too far behind. Looks like 2.0 went live [9 days ago](https://github.com/mozilla-services/syncstorage-rs/issues/348), so not a huge rush, but filing so we've got it.

Can we do the actix upgrade as a standalone task without also converting everything to async/await? (we have a [separate task](https://github.com/mozilla-services/syncstorage-rs/issues/348) filed for async/await changes).

### PR DESCRIPTION
**Pull request by fzzzy**
## Description

This pr updates to actix web 2 and future 0.3, but does not switch anything to async await.

## Testing

Assuming it compiles, cargo test, cargo test in db-tests, and the python tests all pass, it should be good to go.

## Issue(s)

Closes #394

### CODE REVIEW

**Comment by HenkPoley** [comment]
A slight problem is that the Actix project stopped existing at the moment.

**Comment by fzzzy** [comment]
Yes... that is a slight problem...

File: src/db/spanner/support.rs [hunk 1]
```diff
@@ -188,7 +189,13 @@ impl StreamedResultSet {
     ///
     /// Returns false when the stream is finished
     fn consume_next(&mut self) -> Result<bool> {
-        let mut partial_rs = if let Some(result) = self.stream.next() {
+        let (result, stream) = System::new("").block_on(
```
Comment (fzzzy) [inline src/db/spanner/support.rs]: I'm pretty sure the overhead of calling System::new("") on every single bsos request is going to be untenable.

Comment (fzzzy) [reply src/db/spanner/support.rs]: I changed it so that there is a threadlocal System for every thread in the db pool.

**Comment by fzzzy** [comment]
Ok issues are filed for all comments. Merging.

**Pull request merged**